src/realm/object-store/impl/epoll/external_commit_helper.cpp

```diff
@@ -49,31 +49,51 @@ using namespace realm::_impl;
     } while (0)
 
 namespace {
-// Write a byte to a pipe to notify anyone waiting for data on the pipe
-void notify_fd(int fd)
+
+// Make writing to the pipe return -1 when there is no data to read, or no space in the buffer to write to, rather
+// than blocking.
+void make_non_blocking(int fd)
+{
+    int ret = fcntl(fd, F_SETFL, O_NONBLOCK);
+    if (ret == -1) {
+        throw std::system_error(errno, std::system_category());
+    }
+}
+
+// Write a byte to a pipe to notify anyone waiting for data on the pipe.
+// But first consume all bytes in the pipe, since linux may only notify on transition from not ready to ready.
```

Comment:
Maybe add a little not that this opens up for a race condition where the pipe is consumed but the process crashes before it can send the notification?

Reply:
done